components/shared_code/src/shared_data_model/parameters.py

```diff
@@ -52,93 +52,93 @@ class LandingURL(StringParameter):
     This is a string parameter because Quality-time doesn't validate these URLs.
     """
 
-    short_name = "URL"
+    short_name: str = "URL"
 
 
 class SingleChoiceParameter(Parameter):
     """Single choice parameter."""
 
-    type = ParameterType.SINGLE_CHOICE  # noqa: A003
+    type: ParameterType = ParameterType.SINGLE_CHOICE  # noqa: A003
 
 
 class MultipleChoiceParameter(Parameter):
     """Multiple choice parameter."""
 
-    type = ParameterType.MULTIPLE_CHOICE  # noqa: A003
+    type: ParameterType = ParameterType.MULTIPLE_CHOICE  # noqa: A003
     default_value: str | list[str] = []
 
 
 class MultipleChoiceWithAdditionParameter(MultipleChoiceParameter):
     """Multiple choice parameter that allows the user to add additional options."""
 
-    type = ParameterType.MULTIPLE_CHOICE_WITH_ADDITION  # noqa: A003
-    placeholder = "none"
+    type: ParameterType = ParameterType.MULTIPLE_CHOICE_WITH_ADDITION  # noqa: A003
+    placeholder: str = "none"
 
 
 class Username(StringParameter):
     """User to be used for authentication."""
 
     name: str = "Username for basic authentication"
-    short_name = "username"
+    short_name: str = "username"
 
 
 class Password(Parameter):
     """Password parameter."""
 
     name: str = "Password for basic authentication"
-    short_name = "password"
-    type = ParameterType.PASSWORD  # noqa: A003
+    short_name: str = "password"
+    type: ParameterType = ParameterType.PASSWORD  # noqa: A003
 
 
 class PrivateToken(Password):
     """Private token for authentication."""
 
     name: str = "Private token"
-    short_name = "private token"
+    short_name: str = "private token"
     validation_path: str = ""  # URL path to use for the validation of tokens
 
 
 class Days(IntegerParameter):
     """Number of days parameter."""
 
     unit: Unit = Unit.DAYS
-    min_value = "1"
+    min_value: str = "1"
 
 
 class Severities(MultipleChoiceParameter):
     """Security warning severities."""
 
     name: str = "Severities"
-    placeholder = "all severities"
+    placeholder: str = "all severities"
     metrics: list[str] = ["security_warnings"]
 
-    @validator("help_url", always=True)
-    def set_help(cls, help_url: HttpUrl | None, values) -> HttpUrl | None:
+    @model_validator(mode="after")
+    def set_help(self) -> Self:
         """Add a default help string if a help URL was not provided."""
-        if not help_url and not values.get("help"):
-            values["help"] = "If provided, only count security warnings with the selected severities."
-        return help_url
+        if not self.help and not self.help_url:
+            self.help = "If provided, only count security warnings with the selected severities."
+        return self
 
 
 class TestResult(MultipleChoiceParameter):
     """Test result parameter."""
 
     name: str = "Test results"
-    help: str | None = (  # noqa: A003
+    help: str = (  # noqa: A003
```

Comment:
Why is it no longer optional? It still is in `Parameter` and not changed in `MultipleChoiceParameter`?

Reply:
I changed it because mpy complained if I didn't:

```
src/shared_data_model/parameters.py:128: error: Incompatible types in assignment (expression has type "Optional[str]", base class "MultipleChoiceParameter" defined the type as "str")  [assignment]
```

I _think_ mypy is incorrect because MultipleChoiceParameter didn't define the type as "str". It's `str | None` in the base class `Parameter`, but I'm not sure.